src/solvers/dgsem_structured/dg_2d.jl

```diff
@@ -81,9 +81,74 @@ function calc_volume_integral!(du, u,
+    Ja11_node, Ja12_node = get_contravariant_vector(1, contravariant_vectors, i, j, element)
+    Ja21_node, Ja22_node = get_contravariant_vector(2, contravariant_vectors, i, j, element)
+    # diagonal (consistent) part not needed since diagonal of
+    # dg.basis.derivative_split_transpose is zero!
+    set_node_vars!(ftilde1, zero(u_node), equations, dg, i, i, j)
+    set_node_vars!(ftilde2, zero(u_node), equations, dg, j, i, j)
+
+    # contravariant fluxes in the first direction
+    for ii in (i+1):nnodes(dg)
+      u_node_ii = get_node_vars(u, equations, dg, ii, j, element)
+      flux1 = volume_flux(u_node, u_node_ii, 1, equations)
+      flux2 = volume_flux(u_node, u_node_ii, 2, equations)
+      # pull the contravariant vectors and compute their average
+      Ja11_node_ii, Ja12_node_ii = get_contravariant_vector(1, contravariant_vectors, ii, j, element)
+      Ja11_avg = 0.5 * (Ja11_node + Ja11_node_ii)
+      Ja12_avg = 0.5 * (Ja12_node + Ja12_node_ii)
+      # compute the contravariant sharp flux
+      fluxtilde1 = Ja11_avg * flux1 + Ja12_avg * flux2
+      # save and exploit symmetry
+      set_node_vars!(ftilde1, fluxtilde1, equations, dg, i, ii, j)
+      set_node_vars!(ftilde1, fluxtilde1, equations, dg, ii, i, j)
+    end
+
+    # contravariant fluxes in the second direction
+    for jj in (j+1):nnodes(dg)
+      u_node_jj  = get_node_vars(u, equations, dg, i, jj, element)
+      flux1 = volume_flux(u_node, u_node_jj, 1, equations)
+      flux2 = volume_flux(u_node, u_node_jj, 2, equations)
+      # pull the contravariant vectors and compute their average
+      Ja21_node_jj, Ja22_node_jj = get_contravariant_vector(2, contravariant_vectors, i, jj, element)
+      Ja21_avg = 0.5 * (Ja21_node + Ja21_node_jj)
+      Ja22_avg = 0.5 * (Ja22_node + Ja22_node_jj)
+      # compute the contravariant sharp flux
+      fluxtilde2 = Ja21_avg * flux1 + Ja22_avg * flux2
+      # save and exploit symmetry
+      set_node_vars!(ftilde2, fluxtilde2, equations, dg, j,  i, jj)
+      set_node_vars!(ftilde2, fluxtilde2, equations, dg, jj, i, j)
+    end
+  end
+
+  calcflux_twopoint_nonconservative!(ftilde1, ftilde2, u, element,
+                                     have_nonconservative_terms(equations),
+                                     mesh, equations, dg, cache)
+end
+
+
+function calcflux_twopoint_nonconservative!(f1, f2, u, element,
+                                            nonconservative_terms::Val{true},
+                                            mesh::Union{StructuredMesh{2}, UnstructuredMesh2D},
+                                            equations, dg::DG, cache)
+  #TODO: Create a unified interface, e.g. using non-symmetric two-point (extended) volume fluxes
+  #      For now, just dispatch to an existing function for the IdealMhdEquations
+  @unpack contravariant_vectors = cache.elements
+  calcflux_twopoint_nonconservative!(f1, f2, u, element, contravariant_vectors, equations, dg, cache)
+end
```

Comment:
Do you use any of this stuff? I think it was removed when re-designing the nonconservative story of Trixi.jl... Merge conflicts?

Reply:
Yep, there have been merge conflicts. I'll take another look and fix that.